source/crud/compound-operations.txt

```diff
@@ -0,0 +1,52 @@
+===================
+Compound Operations
+===================
+
+.. default-domain:: mongodb
+
+Usually client applications only need to read data out of a database or
+write data into a database. However, client applications sometimes need
+to read and write data in a single interaction with the database. While
+you can accomplish this task using separate reads and writes, doing so
+requires the client application to gracefully handle potential errors
+at any stage of the process and in multiple potential error states.
+This increases the complexity of your code and can make your client
+application brittle and difficult to test.
```

Comment:
- 10/10 suggestion, starting off with "requests" instead of "client applications" makes my intent far clearer to the reader, since client applications generally *do* need to both read and write to the db.
- Good point about ordering, moved that statement.